public/assets/js/dropzoneFunctions.js

```diff
@@ -1,53 +1,110 @@
+                }
+            }
+        },
+        selectFile(selectedFile){
+            try {
+                validationFunctions.validateFile(selectedFile); // validate the file's name, type, and size
+            } catch (error) {
+                this.showError(error.message);
+                return;
+            }
+            publishFileFunctions.previewAndStageFile(selectedFile);
+        },
+        dragenter_handler: function () {
+            var primaryDropzone = document.getElementById('primary-dropzone');
+            primaryDropzone.setAttribute('class', 'dropzone dropzone--drag-over row row--margined row--padded row--tall flex-container--column flex-container--center-center');
+            primaryDropzone.firstElementChild.setAttribute('class', 'hidden');
+            primaryDropzone.lastElementChild.setAttribute('class', '');
 
-function dragover_handler(event) {
-    event.preventDefault();
-}
-
-function dragend_handler(event) {
-    var dt = event.dataTransfer;
-    if (dt.items) {
-        for (var i = 0; i < dt.items.length; i++) {
-            dt.items.remove(i);
+        },
+        dragexit_handler: function () {
+            var primaryDropzone = document.getElementById('primary-dropzone');
+            primaryDropzone.setAttribute('class', 'dropzone row row--tall row--margined row--padded flex-container--column flex-container--center-center');
+            primaryDropzone.firstElementChild.setAttribute('class', '');
+            primaryDropzone.lastElementChild.setAttribute('class', 'hidden');
+        },
+        showError: function (errorMsg) {
+            const errorDisplay = document.getElementById('input-error-file-selection');
+            errorDisplay.hidden = false;
+            errorDisplay.innerText = errorMsg;
+        },
+    },
+    preview: {
+        drop_handler: function (event) {
+            event.preventDefault();
+            var dt = event.dataTransfer;
+            if (dt.items) {
+                if (dt.items[0].kind == 'file') {
+                    var droppedFile = dt.items[0].getAsFile();
+                    this.selectFile(droppedFile);
+                }
+            }
+        },
+        selectFile(selectedFile){
+            this.onmouseleave_handler();
+            try {
+                validationFunctions.validateFile(selectedFile); // validate the file's name, type, and size
+            } catch (error) {
+                this.showError(error.message);
+                return;
+            }
+            publishFileFunctions.previewAndStageFile(selectedFile);
+        },
+        onmouseenter_handler: function () {
+            // show drag-and-drop instructions
+            document.getElementById('asset-preview-dropzone-instructions').setAttribute('class', 'flex-container--column flex-container--center-center position-absolute');
+            // make image lighter
+            document.getElementById('asset-preview').style.opacity = 0.2;
```

Comment:
Could this be done by `:hover` CSS pseudo-class?

Reply:
More generally, I would try to avoid direct style changes in Javascript and instead only perform class modifications. Modifying styles directly means your CSS no longer is sufficient to understand all of the styles inside of the app. If you instead use Javascript to only modify classes and use those classes to apply style changes via CSS, then your CSS still contains all information related to styling.

Reply:
(There may be a few exceptions to "only use classes", like hiding and showing.)